virtcontainers/acrn.go

```diff
@@ -795,22 +795,79 @@ func (a *Acrn) GetMaxSupportedACRNVM() (uint8, error) {
 	return platformInfo.maxKataContainers, nil
 }
 
-func (a *Acrn) storeState() error {
-	if a.store != nil {
-		if err := a.store.Store(store.Hypervisor, a.state); err != nil {
-			a.Logger().WithError(err).Error("failed to store acrn state")
+func (a *Acrn) storeInfo() error {
+	dirPath := VMUUIDStoragePath()
+
+	_, err := os.Stat(dirPath)
+	if os.IsNotExist(err) {
+		// Root directory
+		a.Logger().WithField("path", dirPath).Debugf("Creating UUID directory")
+		if err := os.MkdirAll(dirPath, store.DirMode); err != nil {
 			return err
 		}
+	} else if err != nil {
+		return err
+	}
+
+	dirf, err := os.Open(dirPath)
+	if err != nil {
+		return err
+	}
+	defer dirf.Close()
+
+	if err := syscall.Flock(int(dirf.Fd()), syscall.LOCK_EX|syscall.LOCK_NB); err != nil {
+		return err
+	}
+
+	// write data
+	f, err := os.OpenFile(filepath.Join(dirPath, uuidFile), os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to store information into uuid.json: %v", err)
```

Comment:
Can you use `uuidFile` rather than hard-coding the name of the file in the error (here and below).